pkg/apis/operator/v1alpha1/types_extension.go

```diff
@@ -73,6 +73,7 @@ type ExtensionDeploymentSpec struct {
 	// +optional
 	Values *apiextensionsv1.JSON `json:"values,omitempty"`
 	// RuntimeClusterValues are the deployment values for the extension deployment running in the runtime garden cluster.
+	// If no values are specified, a runtime deployment is considered deactivated.
```

Comment:
TBH, I find this a bit implicit/unexpected. If I just want to run with the default values, I now need to provide a dummy configuration to make sure the extension gets installed to the runtime cluster.

Maybe we should follow the pattern we implemented for seeds/shoots: Only deploy the extension when really needed (i.e., when there are relevant extension objects in the system)? Then we do not have to deal with this question at all.

Reply:
I'll check that 👍

Reply:
I reused parts of [required controller](https://github.com/gardener/gardener/tree/master/pkg/gardenlet/controller/controllerinstallation/required) to decide which extension should be deployed in the runtime cluster.
However, I did not build an own controller for it, but integrated the logic into the extension-controller. We decided a while ago to build one single controller for the `Extension` and I did not want to revert this decision.

There is no `Required` condition or similar yet. Such a condition depends on `ControllerRegistration`/`ControllerDeployment` too and there is an own open point for it in the umbrella issue #9635.